pkg/controller/daemonset/daemonset_controller.go

```diff
@@ -1081,17 +1083,19 @@ func (dsc *ReconcileDaemonSet) hasPodExpectationsSatisfied(ctx context.Context,
 	dsKey := keyFunc(ds)
 	pods, err := dsc.getDaemonPods(ctx, ds)
 	if err != nil {
-		klog.Errorf("Failed to get pods for DaemonSet")
+		klog.ErrorS(err, "Failed to get pods for DaemonSet")
 		return false
 	}
 
 	for _, pod := range pods {
 		dsc.resourceVersionExpectations.Observe(pod)
 		if isSatisfied, unsatisfiedDuration := dsc.resourceVersionExpectations.IsSatisfied(pod); !isSatisfied {
 			if unsatisfiedDuration >= kruiseExpectations.ExpectationTimeout {
-				klog.Errorf("Expectation unsatisfied resourceVersion overtime for %v, wait for pod %v updating, timeout=%v", dsKey, pod.Name, unsatisfiedDuration)
+				klog.ErrorS(fmt.Errorf("expectation unsatisfied resourceVersion overtime for DaemonSet, wait for pod updating"),
```

Comment:
plz just pass nil as error in the ErrorS,

plz change similar logging lines too, pass nil error to ErrorS if no explicit error is available in the context

Reply:
done